File: Chapter 03/JS/3.2 Coupling Exercise/Before/userReportGenerator.js
const UserDataService = require('./userDataService');

class UserReportGenerator {
    constructor() {
        this.dataService = new UserDataService();
    }

    generateReport(userId) {
        const name = this.dataService.getUserName(userId);
        const email = this.dataService.getUserEmail(userId);
        const orders = this.dataService.getUserOrderIds(userId);

        let totalSpent = 0;
        for (const orderId of orders) {
            totalSpent += this.dataService.getOrderTotal(orderId);
        }

        return `User Report for ${name} (${email}) - Total Spent: $${totalSpent.toFixed(2)}`;
    }
}

module.exports = UserReportGenerator;

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/3.2 Coupling Exercise/Before/UserReportGenerator.java
package com.grokkingsoftwarearchitecture.chapter03.couplingexercise.before;

import java.util.List;

public class UserReportGenerator {
    private final UserDataService dataService = new UserDataService();

    public String generateReport(int userId) {
        String name = dataService.getUserName(userId);
        String email = dataService.getUserEmail(userId);
        List<String> orders = dataService.getUserOrderIds(userId);

        double totalSpent = 0.0;
        for (String orderId : orders) {
            totalSpent += dataService.getOrderTotal(orderId);
        }

        return String.format("User Report for %s (%s) - Total Spent: $%.2f", name, email, totalSpent);
    }
}